Development

LucasSwitz opened the pull request:
Merging in the current depth map capturing in C++

Commit:
fixed camera virtual and added depth map

norinew committed:
added header files for calibration and utility

norinew committed:
added calibration's destructor

Commit:
added post process code

Commit:
Merge pull request #2 from cornell-cup/depth_headers

Depth headers

norinew committed:
added calibration matricies

norinew committed:
added remaining code and started the pipeline

Commit:
added disparity matrix creation

norinew committed:
added current code base. memory error when run however...

norinew committed:
merge

norinew committed:
we have ported depth capture to C++ yay!!

ScottSWu commented on calibration_mats/cam_mats_left:
```diff
@@ -0,0 +1,3 @@
+541	0	324
+0	541	222
+0	0	1
```
Were these numbers converted from the npy file? These should not be rounded to integers - the numbers should look something like:

```
calib_left_mat = numpy.array(
    [[ 541.11082257,    0.        ,  324.08789568],
     [   0.        ,  541.11082257,  222.59189031],
     [   0.        ,    0.        ,    1.        ]])
calib_right_mat = numpy.array(
    [[ 541.11082257,    0.        ,  326.00175078],
     [   0.        ,  541.11082257,  231.72506702],
     [   0.        ,    0.        ,    1.        ]])
calib_left_dist = numpy.array(
    [[ 0.09370743,  0.22643711,  0.        ,  0.        , -0.93583831]])
calib_right_dist = numpy.array(
    [[ 0.25088046, -1.03260309,  0.        ,  0.        ,  1.41224436]])
```

ScottSWu commented on src/calibration.cpp:
```diff
@@ -0,0 +1,32 @@
+
+cv::Mat Calibration::readMatFromTxt(std::string filename, int rows, int cols){
+	int m;
+	cv::Mat out = cv::Mat::zeros(rows, cols, CV_32S); //Matrix to store values
```
Following up, shouldn't the datatype be CV_32F

ScottSWu commented on src/constants.h:
```diff
@@ -0,0 +1,20 @@
+#pragma once
```
#ifndef header guard instead of pragma once

ScottSWu commented on src/constants.h:
```diff
@@ -0,0 +1,20 @@
+public:
+	std::vector<std::string> displays = { "Left", "Right", "Dispartiy" };
+
+	static Constants* Instance()
```
C++11 includes smart pointers. Use `std::shared_pointer<Constants>` from `#include <memory>`. Everything else works exactly the same.

ScottSWu commented on src/constants.h:
```diff
@@ -0,0 +1,20 @@
+class Constants
+{
+public:
+	std::vector<std::string> displays = { "Left", "Right", "Dispartiy" };
```
"Disparity" spelling

ScottSWu commented on src/display.h:
```diff
@@ -17,7 +17,7 @@ class Display {
     virtual ~Display();
 
-    void displayImages(std::vector<cv::Mat>& displayImages);
+    void displayImages(std::vector<cv::Mat*>& displayImages);
```
If the vector is being passed by reference, I'm not sure the matrices need to be pointers anymore to prevent copying.

ScottSWu commented on src/opencvcamera.h:
```diff
@@ -0,0 +1,32 @@
+	//caller responsible for cleaning up each capture
+	cv::Mat* Capture()
+	{
+		cv::Mat* mat = new cv::Mat();
```
One thing you can do here instead of using pointers is pass in the output matrix to the function.

```
void Capture(cv::Mat& output) {
    _capture >> output;
}
```

ScottSWu commented on src/opencvcamera.h:
```diff
@@ -0,0 +1,32 @@
+#pragma once
```
Header guard?

ScottSWu commented on src/stereorig.h:
```diff
@@ -0,0 +1,35 @@
+#pragma once
```
Header guard?

ScottSWu reviewed (commented):
Overall, try to use fewer raw pointers, either by returning by parameter reference, or using smart pointers if you really need to.

Commit:
added code fixes and cleanup and comments

LucasSwitz committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into development

LucasSwitz committed:
Merge branch 'development' of https://github.com/cornell-cup/cs-slam-core into development

LucasSwitz committed:
Refactoring and Bug Fixed

- Encapsulated single camera funcitonality in single class
- Made Display a Singleton that now operates on sets rather than
vectors.
- add draw() to Display to simplify image projecting
- Added StereoCapture struct to better encapsulate a single Stereo Image
Group along with a timestamp.

norinew committed:
added new calibration files

norinew committed:
Merge branch 'development' of https://github.com/cornell-cup/cs-slam-core into development

LucasSwitz committed:
added opencv.cpp

LucasSwitz committed:
moved file into src

Commit:
removed necessary files and refactored some code

norinew committed:
fixed compile erros and added point cloud

norinew committed:
finished point cloud

norinew committed:
added nudging

norinew committed:
removed point cloud

norinew committed:
created disparity pipeline class

ScottSWu commented on src/main.cpp:
```diff
@@ -78,62 +12,14 @@ int main() {
-	cv::namedWindow("left");
-	cv::namedWindow("right");
-	cv::namedWindow("disparity");
+	DisparityPipeline pipeline = DisparityPipeline(leftCamera, rightCamera);
```
This can be declared with
`DisparityPipeline pipeline(leftCamera, rightCamera);`
